kafka-1.0.x/src/main/scala/monix/kafka/KafkaConsumerObservableManualCommit.scala

```diff
@@ -0,0 +1,115 @@
+/*
+ * Copyright (c) 2014-2019 by The Monix Project Developers.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package monix.kafka
+
+import monix.eval.Task
+import monix.execution.Ack.Stop
+import monix.execution.cancelables.BooleanCancelable
+import monix.execution.{Ack, Callback}
+import monix.reactive.Observer
+import monix.reactive.observers.Subscriber
+import org.apache.kafka.clients.consumer.{KafkaConsumer, OffsetAndMetadata, OffsetCommitCallback}
+import org.apache.kafka.common.TopicPartition
+
+import scala.collection.JavaConverters._
+import scala.concurrent.{blocking, Future}
+import scala.util.control.NonFatal
+import scala.util.{Failure, Success}
+
+/** KafkaConsumerObservable with ability to manual commit offsets
+  * and forcibly disables auto commits in configuration.
+  * Such instances emit [[CommittableMessage]] instead of Kafka's ConsumerRecord.
+  */
+final class KafkaConsumerObservableManualCommit[K, V] private[kafka] (
+  override protected val config: KafkaConsumerConfig,
+  override protected val consumer: Task[KafkaConsumer[K, V]])
+    extends KafkaConsumerObservable[K, V, CommittableMessage[K, V]] {
+
+  // Caching value to save CPU cycles
+  private val pollTimeoutMillis = config.fetchMaxWaitTime.toMillis
+
+  override protected def ackTask(consumer: KafkaConsumer[K, V], out: Subscriber[CommittableMessage[K, V]]): Task[Ack] =
+    Task.create { (scheduler, cb) =>
+      implicit val s = scheduler
+      val asyncCb = Callback.forked(cb)
+      val cancelable = BooleanCancelable()
+
+      val commit: Commit = new Commit {
```

Comment:
We could probably create it outside this method and reuse

Reply:
We need `KafkaConsumer[K, V]` in scope for `Commit` creation.

I can move it out to `private def commit(consumer: KafkaConsumer[K, V]): Commit = ???`, but I have no ideas how to create it once and reuse later.

Reply:
Ah, right, I've missed it. I'll let you know if anything comes to my mind